core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/ProgressiveUtils.scala

```diff
@@ -0,0 +1,38 @@
+package edu.uci.ics.texera.workflow.common
+
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, AttributeType}
+
+object ProgressiveUtils {
+
+  // boolean attribute to indicate insertion / retraction
+  // true  indicates insertion  (+)
+  // false indicates retraction (-)
+  val insertRetractFlagAttr = new Attribute("__internal_is_insertion", AttributeType.BOOLEAN)
```

Comment:
So for now the insert/retract flag is an attribute in the tuple schema, should we make it a part of the tuple itself in the future?

Reply:
I also think it should be an attribute of `Tuple` since we are decoupling `Schema` from `Tuple`.

Reply:
I think Flink includes the flag as part of the tuple, rather than as one data column of the tuple. I believe both approaches have pros and cons. 

I think the current way is good for 2 reason: 1) it makes minimal changes to the existing codebase, and 2) the flag is treated uniformly as any other data. I believe such uniformity is good because this flag, just as any other columns, can be used in computation. We'll keep this format for now.